File: grailsapp/src/java/s/Service.java
package s;

import java.util.ArrayList;
import java.util.List;


public class Service {
	private static List<Product> products = new ArrayList<Product>();
	private static List<Category> categories = new ArrayList<Category>();;
	
	static {
		System.out.println("Creating products");
		
		for (int i = 0; i < 5; i++) {
			String name = Integer.toString(1000 + i);
			categories.add(new Category(name));
		}
		
		for (int i = 0; i < 5000; i++) {
			String name = Integer.toString(i);
			String description = Integer.toString(i * i);
			
			Product product = new Product(name, i, description);
			
			product.getCategories().addAll(categories);
			products.add(product);
		}
	}
	
	public static List<Product> getProducts() {
		return products;
	}

}

File: aspnetmvc-3/Models/Service.cs
namespace mvcapp.Models
{
    using System.Collections.Generic;

    public class Service
    {
        #region Constants and Fields

        private static readonly List<Category> categories = new List<Category>();

        private static readonly List<Product> products = new List<Product>();

        #endregion

        #region Constructors and Destructors

        static Service()
        {
            for (var i = 0; i < 5; i++)
            {
                string name = (1000 + i).ToString();
                categories.Add(new Category(name));
            }

            for (var i = 0; i < 20000; i++)
            {
                string name = i.ToString();
                string description = (i * i).ToString();

                var product = new Product(name, i, description);

                product.Categories.AddRange(categories);
                products.Add(product);
            }
        }

        #endregion

        #region Public Methods

        public static List<Product> GetProducts(int max)
        {
            return products.GetRange(0, max);
        }

        #endregion
    }
}